### PR TITLE
Moretasks

### PR DESCRIPTION
**Pull request by joaquinvanschoren**
#### What does this PR implement/fix? Explain your changes.

It generalizes tasks so that you can handle different tasks differently.
Right now every task was assumed to be a classification task

#### How should this PR be tested?
Standard unit test

### CODE REVIEW

File: openml/tasks/task.py [hunk 1]
```diff
@@ -7,38 +7,58 @@
...
-    def __init__(self, task_id, task_type_id, task_type, data_set_id,
-                 target_name, estimation_procedure_type, data_splits_url,
-                 estimation_parameters, evaluation_measure, cost_matrix,
-                 class_labels=None):
+    def __init__(self, task_id, task_type_id, task_type, data_set_id, estimation_procedure_type,
+                 estimation_parameters, evaluation_measure):
         self.task_id = int(task_id)
         self.task_type_id = int(task_type_id)
         self.task_type = task_type
         self.dataset_id = int(data_set_id)
```
Comment (janvanrijn) [inline openml/tasks/task.py]: can we call this "source_data", in co-ordinance with the task descriptions (hidden in the XML)

File: openml/tasks/task.py [hunk 2]
```diff
@@ -7,38 +7,58 @@
...
-        self.target_name = target_name
         self.estimation_procedure = dict()
         self.estimation_procedure["type"] = estimation_procedure_type
-        self.estimation_procedure["data_splits_url"] = data_splits_url
         self.estimation_procedure["parameters"] = estimation_parameters
         #
         self.estimation_parameters = estimation_parameters
```
Comment (janvanrijn) [inline openml/tasks/task.py]: I would rather store the estimation procedure id (not available in api right now, let's add it)

File: openml/tasks/task.py [hunk 3]
```diff
@@ -7,38 +7,58 @@
...
-        self.split = None
-
-        if cost_matrix is not None:
-            raise NotImplementedError("Costmatrix")
 
     def get_dataset(self):
         """Download dataset associated with task"""
         return datasets.get_dataset(self.dataset_id)
 
+    def push_tag(self, tag):
```
Comment (janvanrijn) [inline openml/tasks/task.py]: didn't we have general tag functions?

Comment (janvanrijn) [reply openml/tasks/task.py]: Otherwise we should create them. (and not duplicate tag functions code for each entity)

Comment (mfeurer) [reply openml/tasks/task.py]: Great remark. I think we should do this in a new PR, though, as it is not directly related to this PR. I opened issue #549 for this.

File: openml/tasks/task.py (hunk 2 above)
Comment (mfeurer) [reply openml/tasks/task.py]: @janvanrijn I'm afraid that won't be backwards compatible... But we can probably transit at some point to store something different in the cache. Do you want to open a new issue on the OpenML issue tracker?

**Pull request merged**